addons/spreadsheet/static/src/global_filters/helpers.js

```diff
@@ -696,22 +861,22 @@ function getNextRelativeDateFilterValue(value) {
  * @param {RelativeDateValue} value
  * @returns {RelativeDateValue}
  */
-function getPreviousRelativeDateFilterValue(value) {
-    switch (value.period) {
+function getPreviousRelativeDateFilterValue(period) {
```

Comment:
the docstring should be updated :) + it's false, it returns a `DateValue` not a `RelativeDateValue`

The function name might be misleading?